src/main/java/spireMapOverhaul/zones/gremlinTown/monsters/GremlinCannon.java

```diff
@@ -0,0 +1,141 @@
+package spireMapOverhaul.zones.gremlinTown.monsters;
+
+import basemod.abstracts.CustomMonster;
+import com.badlogic.gdx.graphics.g2d.SpriteBatch;
+import com.megacrit.cardcrawl.actions.AbstractGameAction;
+import com.megacrit.cardcrawl.actions.common.ApplyPowerAction;
+import com.megacrit.cardcrawl.actions.common.RollMoveAction;
+import com.megacrit.cardcrawl.actions.utility.TextAboveCreatureAction;
+import com.megacrit.cardcrawl.core.CardCrawlGame;
+import com.megacrit.cardcrawl.core.Settings;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+import com.megacrit.cardcrawl.localization.MonsterStrings;
+import com.megacrit.cardcrawl.powers.VulnerablePower;
+import spireMapOverhaul.SpireAnniversary6Mod;
+import spireMapOverhaul.zones.gremlinTown.events.Surprise;
+import spireMapOverhaul.zones.gremlinTown.events.misc.Shell;
+
+import static spireMapOverhaul.util.Wiz.*;
+
+public class GremlinCannon extends CustomMonster
```

Comment:
Funny enemy! I think it's reasonably balanced given the fight it shows up in. Big debuffs, but taking every other turn off.